ocaml/libs/http-lib/http.ml

```diff
@@ -1018,32 +1018,27 @@ module Url = struct
     let params = if params = [] then "" else "?" ^ kvpairs params in
     uri ^ params
 
-  (* Wrap a literal IPv6 address in square brackets; otherwise pass through *)
-  let maybe_wrap_IPv6_literal addr =
-    if Unixext.domain_of_addr addr = Some Unix.PF_INET6 then
-      "[" ^ addr ^ "]"
-    else
-      addr
-
   let to_string = function
     | File {path}, data ->
-        Printf.sprintf "file:%s%s" path (data_to_string data) (* XXX *)
-    | Http h, data ->
-        let userpassat =
+        (* this should have file:// *)
+        Printf.sprintf "file:%s%s" path (data_to_string data)
```

Comment:
I believe Uri can be used here as well, and in all cases, hopefully.

I'd try to generate the parameters for `Uri.make`:
```ocaml
let scheme, host, port, userinfo, path, query = match url with
  | File {path}, data ->
       "file", None, None, None, (Printf.sprintf "file:%s%s" path (data_to_string data)), None
...
in
Uri.(make ~scheme ~host ~port ~userinfo ~path ~query |> to_string)

Reply:
The code above does not work because `scheme`, `host` and so on become shadowed once Uri` is open:
```
utop # let host = "localhost" in Uri.(make ~host () |> to_string);;
Error: This expression has type t -> string option
       but an expression was expected of type string
```